File: OOPs/Composition/composition.py
class Engine:
    def start(self):
        print("Engine starting...")

class Car:
    def __init__(self):
        self.engine = Engine()  # Composition: Car "has-a" Engine

    def start(self):
        self.engine.start()  # Delegating functionality to Engine
        print("Car is ready to drive!")

car = Car()
car.start()

File: OOPs/Composition/composition.cpp
#include<bits\stdc++.h>
using namespace std;

class Engine {
public:
    void start() {
        cout << "Engine starting..." << endl;
    }
};

class Car {
private:
    Engine engine;  // Composition: Car "has-a" Engine

public:
    void start() {
        engine.start();  // Delegating functionality to Engine
        cout << "Car is ready to drive!" << endl;
    }
};

int main() {
    Car car;
    car.start();
    return 0;
}